src/EppoClient.php

```diff
@@ -506,23 +516,27 @@ private function getBanditDetail(
     ): BanditResult {
         Validator::validateNotBlank($flagKey, 'Invalid argument: flagKey cannot be blank');
 
+        $config = $this->configurationStore->getConfiguration();
+
         try {
-            $variation = $this->getStringAssignment(
+            $variation = $this->getAssignmentDetail(
                 $flagKey,
                 $subjectKey,
                 $subject->toArray(),
-                $defaultValue
-            );
+                VariationType::STRING,
+                $config
+            )?->key ?? $defaultValue;
+            // TODO return a BanditResult with the default value here instead of going with the default.
```

Comment:
Does this basically happen anyways on line 574? (agree it would be easier to understand if we just did it right here)